### DISCUSSION TITLE
Worldborder

### DISCUSSION THREAD

**Discussion by object-Object** [HexxyCraft/modpack#12] [General]
How big should the server's worldborder be?

Things to consider:
* We can increase it later if we're running out of space or want to add new worldgen mods
* If we pregenerate chunks, smaller worldborders will be easier

My suggestion is 8000 blocks (diameter).

**Comment by maraxin** [comment]
My suggest is that it should be between 5000x5000 - 8000x8000. This is server after all, so it would be nice to have lower worldborder (looting won't be the problem with botania orechid and lootr)

**Comment by beholderface** [comment]
Not like a relatively large (16k-24k across) worldborder would be a big obstacle to multiplayer interaction, with GTP existing. Though I do agree with the points Object made in the initial post.

**Comment by LlamaSageR** [comment]
5000x5000 would be the best in my opinion. We can always expand the world border as needed, but you can't *un*expand a world border.

**Comment by object-Object** [comment]
From Adkkid on Discord:
> my two cents on world border: its significantly better to start with a smaller worldborder, preload chunks, then expand as needed, compared to making it too big too soon.
> My go-to strategy is to start it off very small around ~1k-2k diameter to encourage players to settle closer together/interact, but then have it set to gradually expand over the first week or two of the server, usually around 5k diameter. But that might be less relevant since we'll have GTP/gates 
> on one kitchen sink type server, we kept a [2.5k] diameter world border the whole time and never ran into significant issues - we found it heavily encouraged interaction
> Playerbase fluctuated a ton - At peak, it was probably only ~20 people who were logging on regularly
> http://server.galaxy-craft.com:8123/

**Comment by LlamaSageR** [comment]
A 2.5k diameter world sounds nice.

**Comment by object-Object** [comment]
Proposal based on the above suggestions:

Start with a 2k block worldborder, but pregenerate out to some limit (8k?). At some automated interval (every week? every few days?) increase the border by some amount (1k? 2k?) until we reach the limit.

Thoughts?

**Reply by LlamaSageR** [reply]
I think having it at 2k and manually expanding it based on demand would be better. Helps make sure it is as small as it can be.

**Reply by autumn-birds** [reply]
idk how hard the "automated" part would be-- and you might want it a human in the loop for something as consequential as world border changes-- but maybe starting small and expanding by say, 500 blocks a week for awhile until target size, could bring some incentive and interest to keep coming back to the server?

i like this thought mainly because it seems like alot of servers die off due to inactivity, so anything event-like that could change things up feels like a _potential_ win.  and i like having a smallish/constrained target size (not TOO small ofc, but still) because it seems like it makes the world more... personal?  it seems like a different vibe than when you can just walk forever.

that said this all might be "technical solution to a social/cultural problem" bogus.  idk

**Reply by LlamaSageR** [reply]
I guess my main consideration is that there isn't much room to expand too. Having the world be small helps it be more connected, helps with the filesize / performance, and makes it overall be more lively. While there being some sort of event is good, having the world border procedurally expand would either ruin that small server size, or the expansion itself would be inconsequential. Afterall, if the world starts small, and ends up small, nothing really changes (Unless the world is *too* small to begin with, but that's an even worse issue).

That's why I suggest that the world be expanded a necessary, rather than be some server event. (Though if there is some need for it, it could absolutely be made an event! The event shouldn't be the primary reason for doing it though.)

**Reply by object-Object** [reply]
How about we start at 2k and expand to 4k or 5k after a week, then leave it as-is unless there's some reason to go further? So there's solid encouragement to settle nearby, but we get more room to find stuff later. I think that strikes a balance between what y'all are saying.

**Reply by LlamaSageR** [reply]
Assuming it's Diameter, I'd say that'd work. 1000 out in each direction is small at the beginning to encourage close interaction, and then expanding to 2k-2.5k gives more room for later.

**Comment by object-Object** [comment]
Yes, diameter, since that's what the worldborder command uses.
